File: resources/recipes/authentication/password-hashing/hash_argon2.js
const argon2 = require('argon2');

const password = 'supersecret';

async function main() {
  const start = Date.now();
  const hash = await argon2.hash(password, {
    type: argon2.argon2id,
    memoryCost: 65536,
    timeCost: 3,
    parallelism: 1
  });
  const hashMs = Date.now() - start;
  console.log(`Argon2id hash: ${hash}`);
  console.log(`hash time: ${hashMs} ms`);
  const ok = await argon2.verify(hash, password);
  console.log(`verify: ${ok}`);
}

main().catch(console.error);

File: resources/recipes/authentication/password-hashing/hash_argon2.py
import time
from argon2 import PasswordHasher


def hash_password(password: str) -> str:
    ph = PasswordHasher(time_cost=3, memory_cost=65536, parallelism=1)
    return ph.hash(password)


def verify_password(hashed: str, password: str) -> bool:
    ph = PasswordHasher(time_cost=3, memory_cost=65536, parallelism=1)
    try:
        ph.verify(hashed, password)
        return True
    except Exception:
        return False


if __name__ == "__main__":
    password = "supersecret"
    start = time.perf_counter()
    hashed = hash_password(password)
    elapsed_ms = (time.perf_counter() - start) * 1000
    print(f"Argon2id hash: {hashed}")
    print(f"hash time: {elapsed_ms:.2f} ms")
    print(f"verify: {verify_password(hashed, password)}")
